### ISSUE TITLE
Track game data server side

### ISSUE THREAD

**Issue by timotheeg** [timotheeg/nestrischamps#34]
At the moment, games are reported from the UI, since UI computes all the game stats. 

But because we have 2 different types of UI with different code (historical reason), it's a pain to maintain game reporting code. 

Also, in match layouts, the layout is owned by one user (the game host), but may host 2 different players, and it's not right to have the game host report the game data from different users, from an security perspective, it implies there's an API where a user may report for another, and I don't like that. 

We could add a column "reporter" in the DB, so we could still sort of clean bad entries if we wanted, but basically, I thin the solution is to track the game progression in the server, straight from the producer frames.

On the server, there's one ongoing game for a producer, tracking the minimum amount of stats we want to capture in DB. For reference, at the moment, the score table has this structure:

```
    Column    |            Type             | Collation | Nullable |              Default
--------------+-----------------------------+-----------+----------+------------------------------------
 id           | integer                     |           | not null | nextval('scores_id_seq'::regclass)
 datetime     | timestamp without time zone |           | not null |
 player_id    | bigint                      |           | not null |
 start_level  | smallint                    |           |          |
 end_level    | smallint                    |           |          |
 score        | integer                     |           |          |
 lines        | smallint                    |           |          |
 tetris_rate  | double precision            |           |          |
 num_droughts | smallint                    |           |          |
 max_drought  | smallint                    |           |          |
 das_avg      | double precision            |           |          |
 duration     | integer                     |           |          | 0
 clears       | character varying(400)      |           |          | ''::character varying
 pieces       | character varying(1200)     |           |          | ''::character varying
 transition   | integer                     |           |          |
Indexes:
    "scores_pkey" PRIMARY KEY, btree (id)
```

So basically, on the connection, a Game instance should run, and intercept game frames, compute basic stats (e.g. TRT and num droughts) like the UI currently does locally, and record the game into the DB when game over event is detected.

This will allo ALL games to be tracked, even the ones done in competition.

BONUS: the server could store all the frame entries into a binary file in S3 or digital ocean, such that any game could be replayed

**Comment by timotheeg** [comment]
Done in https://github.com/timotheeg/nestrischamps/pull/37
